plugins/woocommerce/src/Admin/Features/ProductBlockEditor/RedirectionController.php

```diff
@@ -155,6 +155,13 @@ protected function get_parsed_route(): array {
 		$path        = esc_url_raw( wp_unslash( $_GET['path'] ) );
 		$path_pieces = explode( '/', wp_parse_url( $path, PHP_URL_PATH ) );
 
+		if ( 3 !== count( $path_pieces ) ) {
+			return array(
+				'page'       => null,
```

Comment:
I see how this will make the error gone, but are we sure about this change? As it's not 1 to 1 replacement of the previous code., so it might result in unintended consequences. Unless it was also buggy.

Previous code would always return `'page'       => $path_pieces[1],` if the count of `$path_pieces` is at least 2. New code won't get there if count of `$path_pieces` is 2. 

`product_id` will work as expected

Reply:
Would appreciate another 👀 after 3be805b

Reply:
This change looks like it should work. I'll be able to retest it a later today, if needed.